.changeset/many-lobsters-accept.md

```diff
@@ -0,0 +1,8 @@
+---
+"astro": minor
+---
+
+Fixes an issue with view transition names containing spaces or punctuation.
+
+This fix could be a breaking change if you leverage details about how Astro translates `transition:name` directives into values of the underlying CSS `view-transition-name` property.
+This mainly affects spaces and punctuation marks but no unicode characters with codes >= 128.
```

Comment:
```suggestion
Regular use naming your transitions with `transition: name` is unaffected.

However, this fix may result in breaking changes if your project relies on the particular character encoding strategy Astro uses to translate `transition:name` directives into values of the underlying CSS `view-transition-name` property. For example `Welcome to Astro` is now encoded as `Welcome20_to20_Astro2e_`.

This mainly affects spaces and punctuation marks but no Unicode characters with codes >= 128.
```

OK, this may be a little much, and of course, may not be entirely accurate. But, would something like this be helpful? It is a minor, and this *might* affect people, so I grabbed some context from the error and PR description in case we liked this.

Reply:
Perfect! Thanks a lot!